### DISCUSSION TITLE
[Config Support]: zone restrictions

### DISCUSSION THREAD

**Discussion by turboc1208** [blakeblackshear/frigate#10353] [Config Support]
### Describe the problem you are having

In the docs there is a section where it talks about a cat being in zone 1 but not zone 2 because the bottom center of the bounding box is in zone 1 but not zone 2.  If I want to record cats walk through the "T" intersection of the sidewalk and the home path, Do I need more than one zone defined.  Could I just specify a zone in that intersection and say it's required.  Once the target leaves that zone, it would be followed until it leaves the frame.  Right?
Second, is there a better place to post these types of questions?

### Version

0.13.2-6476F8A

### Frigate config file

```yaml
NA
```

### Relevant log output

```shell
NA
```

### Frigate stats

```json
NA
```

### Operating system

Debian

### Install method

Docker Compose

### Coral version

USB

**Comment by NickM-27** [comment]
You just need to define one zone for the area that you care about. Sometimes you want to handle different areas separately (like only care about cars in the driveway) in which case multiple required zones can be used. 

Yes, the clip will include the entire time the object was detected.

**Comment by turboc1208** [comment]
The problem is the car becomes stationary and it never stops recording.  I had an 8 hour recording this morning of my wife’s car sitting in the driveway.  Once the car becomes stationary, I want it to stop recording the car.  If a person, dog, cat is detected around the car, record that until it leaves the area, but once the car stops, stop recording it.

Right now, I’ve got the street setup as a required zone.  We live in a cove so there isn’t much through traffic.  I want to see all the cars coming and going and once they pull into my driveway and stop, stop recording them until someone gets in/out of them or they pull back out into the street.

From: Nicolas Mowen ***@***.***>
Date: Saturday, March 9, 2024 at 11:33 AM
To: blakeblackshear/frigate ***@***.***>
Cc: turboc1208 ***@***.***>, Author ***@***.***>
Subject: Re: [blakeblackshear/frigate] [Config Support]: zone restrictions (Discussion #10353)

You just need to define one zone for the area that you care about. Sometimes you want to handle different areas separately (like only care about cars in the driveway) in which case multiple required zones can be used.

Yes, the clip will include the entire time the object was detected.

**Reply by NickM-27** [reply]
What makes you think it is recording? Just because the event is ongoing doesn't mean it is recording.

**Comment by turboc1208** [comment]
There was a 8 hour event (not ongoing) when I looked back to see what was chewing up half of my disk space.  LOL

From: Nicolas Mowen ***@***.***>
Date: Saturday, March 9, 2024 at 3:05 PM
To: blakeblackshear/frigate ***@***.***>
Cc: turboc1208 ***@***.***>, Author ***@***.***>
Subject: Re: [blakeblackshear/frigate] [Config Support]: zone restrictions (Discussion #10353)

What makes you think it is recording? Just because the event is ongoing doesn't mean it is recording.

**Reply by NickM-27** [reply]
Right but still, just because the event was 8 hours in length doesn't mean it recorded all of that was recorded. This depends on your recording retention config

**Comment by turboc1208** [comment]
So what do the numbers of seconds in parentheses after the time ago mean if that’s not how much was recorded?  Are you saying there are multiple recording segments compressed into one event ignoring the time when the car was stationary?

[A screenshot of a dog  Description automatically generated]

From: Nicolas Mowen ***@***.***>
Date: Saturday, March 9, 2024 at 3:13 PM
To: blakeblackshear/frigate ***@***.***>
Cc: turboc1208 ***@***.***>, Author ***@***.***>
Subject: Re: [blakeblackshear/frigate] [Config Support]: zone restrictions (Discussion #10353)

Right but still, just because the event was 8 hours in length doesn't mean it recorded all of that was recorded. This depends on your recording retention config

**Reply by NickM-27** [reply]
It's the amount of time that the vehicle was detected and tracked. 

Yes, as the https://docs.frigate.video/configuration/record explain recordings are saved in 10 seconds segments and dynamically joined together to display events and continuous recordings (depending on your config)

**Comment by turboc1208** [comment]
Ok, I think I get it now.  Assuming I have record/retain/mode=motion and record/events/retail/mode=active_objects then; The event is the total time it’s tracking the object until it leaves.  During that time, there may be motion by the object, or other objects may come in and leave from the frame.  Those motion events are recorded and kept as part of the event.  When everything is quiet, the event keeps watching the object and just doesn’t record what it sees.  Right?

From: Nicolas Mowen ***@***.***>
Date: Saturday, March 9, 2024 at 3:34 PM
To: blakeblackshear/frigate ***@***.***>
Cc: turboc1208 ***@***.***>, Author ***@***.***>
Subject: Re: [blakeblackshear/frigate] [Config Support]: zone restrictions (Discussion #10353)

It's the amount of time that the vehicle was detected and tracked.

Yes, as the https://docs.frigate.video/configuration/record explain recordings are saved in 10 seconds segments and dynamically joined together to display events and continuous recordings (depending on your config)

**Reply by NickM-27** [reply]
Yes you have it right.

To be clear we realize this isn't optimal for how to review footage as you don't actually care about when the car is parked and not moving. The next version of frigate will have a new UI and a new way of presenting this information

**Comment by turboc1208** [comment]
Any love being given to the configuration?  Maybe making it option driven through a gui instead of a text file?  I keep getting confused on what restrictions I can put where whether things can be done at the camera level or have to be done at the global level. It might make it a bit more intuitive for those of us that haven’t been with frigate a long time yet.

And thanks for you help today.


From: Nicolas Mowen ***@***.***>
Date: Saturday, March 9, 2024 at 4:56 PM
To: blakeblackshear/frigate ***@***.***>
Cc: turboc1208 ***@***.***>, Author ***@***.***>
Subject: Re: [blakeblackshear/frigate] [Config Support]: zone restrictions (Discussion #10353)

Yes you have it right.

To be clear we realize this isn't optimal for how to review footage as you don't actually care about when the car is parked and not moving. The next version of frigate will have a new UI and a new way of presenting this information

**Reply by NickM-27** [reply]
Eventually, not for the next update, it will be a long process. 

Almost all options can be set at the camera level.
